Automattic/pocket-casts-android
# Task/eoy 2024 screenshots

## Description

This adds screenshot detection to the PB24 flow. Only sharable stories are acted upon to show the dialog. This is only the UI part. Actual sharing from the dialog will be done once I implement sharing.

> [!note]
> This work reliably only on SDK 34+ because there are official APIs. On SDK below 34 it depends on the naming convention of screenshots. 

Figma: lH66LwxxgG8btQ8NrM0ldx-fi-3070_26832

## Testing Instructions

1. Launch PB24.
2. Take a screenshot of the Cover story.
3. Ensure that no dialog appears.
4. Navigate to the Number of Shows story.
5. Take a screenshot of this story.
6. Verify that a dialog appears and the story auto-progress pauses.
7. Tap any button on the dialog.
8. Confirm that the dialog disappears and the story auto-progress resumes.

## Screenshots or Screencast 

https://github.com/user-attachments/assets/0f4e31ae-0e81-4160-9a22-1b0f5b2327e0

## Checklist
 
#### I have tested any UI changes...
